File: typescript/examples/auth-client-credentials.ts
import { ClientCredentialsTokenSource, TwoFinanceClient, configFromEnv } from "../src/index";

async function main() {
  const tokenSource = new ClientCredentialsTokenSource({
    tokenURL: process.env.TWO_FINANCE_AUTH_TOKEN_URL || "",
    clientID: process.env.TWO_FINANCE_AUTH_CLIENT_ID || "",
    clientSecret: process.env.TWO_FINANCE_AUTH_CLIENT_SECRET || "",
    scopes: ["2finance.sdk"]
  });

  const client = new TwoFinanceClient({
    ...configFromEnv(process.env),
    tokenSource
  });
  console.log(await client.analytics.indicators());
}

main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});

File: python/examples/auth_client_credentials.py
import os

from twofinance_sdk_client import ClientCredentialsTokenSource, TwoFinanceClient, config_from_env


def main() -> None:
    token_source = ClientCredentialsTokenSource(
        token_url=os.environ.get("TWO_FINANCE_AUTH_TOKEN_URL", ""),
        client_id=os.environ.get("TWO_FINANCE_AUTH_CLIENT_ID", ""),
        client_secret=os.environ.get("TWO_FINANCE_AUTH_CLIENT_SECRET", ""),
        scopes=["2finance.sdk"],
    )
    client = TwoFinanceClient(config_from_env(), token_source=token_source)
    print(client.analytics.indicators())


if __name__ == "__main__":
    main()
